test/src/containerTestUtils.ts

```diff
@@ -0,0 +1,105 @@
+ *-------------------------------------------------------------------------------------------------------------*/
+
+import * as fs from 'fs';
+import * as path from 'path';
+import * as https from 'https';
+import * as Docker from 'dockerode';
+import * as config from '../config/testConfig.json';
+
+let logLevel: string;
+
+export function getConfig(property: string, defaultVal: any = undefined) {
+	// Generate env var name from property - pascalCase to PASCAL_CASE
+	const envVar = property.split('').reduce((prev: string, next: string) => {
+		if (next >= 'A' && next <= 'Z') {
+			return prev + '_' + next;
+		} else {
+			return prev + next.toLocaleUpperCase();
+		}
+	}, '');
+
+	return process.env[envVar] || (<any>config)[property] || defaultVal;
+}
+
+export async function getStringFromUri(uri: string) {
+	return new Promise<string>((resolve, reject) => {
+		https.get(uri, (res) => {
+			let result = '';
+			res.on('data', (chunk)=> { result += chunk.toString(); })
+			res.on('close',() => { resolve(result); });
+			res.on('error', (err) => { reject(err); });
+		});	
+	})
+}
+
+export function getLogLevel() {
+	if (!logLevel) {
+		logLevel = getConfig('testLogLevel', 'info');
+	}
+	return logLevel;
+}
+
+export function log(level: string, message: string) {
+	let levels: string[] = [];
+	switch (getLogLevel()) {
+		case 'error': levels = ['error']; break;
+		case 'info': levels = ['error', 'info']; break;
+		case 'debug': levels = ['error', 'info', 'debug']; break;
+		case 'trace': levels = ['error', 'info', 'debug', 'trace']; break;
+	}
+	if (levels.indexOf(level) >= 0) {
+		console.log(message);
+	}
+}
+
+export function getRootFolderRealPath(type: string) {
+	const rootPath = getConfig('devContainerRoot', path.join(__dirname, '..', '..'));
+	const expectedDevConPath = path.join(rootPath, type);
+	return fs.existsSync(expectedDevConPath) ? fs.realpathSync(expectedDevConPath) : expectedDevConPath;
+}
+
+export function getDefinitionListFromPath(definitionsPath: string, configProperty: string, getFn: (path: string) => {} = defaultGetDefnitionListFn) {
+	const subsetToTest = getConfig(configProperty);
+	if (subsetToTest) {
+		if (subsetToTest === 'none') {
+			return [];
+		}
+		if (typeof subsetToTest === 'string') {
+			return [subsetToTest];
+		} else {
+			return subsetToTest;
+		}
+	} else {
+		return getFn(definitionsPath);
+	}
+}
+
+function defaultGetDefnitionListFn(definitionsPath: string) {
+	const definitionList: string[] = [];
+	fs.readdirSync(definitionsPath).forEach((dir: string) => {
+		if (fs.statSync(path.join(definitionsPath, dir)).isDirectory()) {
+			definitionList.push(dir);
+		}
+	});
+	return definitionList;
+
+}
+
+
+export function findMountDestination(containerInfo: Docker.ContainerInspectInfo, sourceDir: string, vscodeDevConPath: string) {
+	for (let i = 0; i < containerInfo.Mounts.length; i++) {
+		const mount = containerInfo.Mounts[i];
```

Comment:
Since you're using typescript and latest ES, you could shorten this with a nice `for..of`  😄 
<pre>
for (const mount of containerInfo.Mounts) {
</pre>